src/os_common/os_fhandle.c

```diff
@@ -240,6 +241,20 @@ __wt_open(WT_SESSION_IMPL *session,
 	WT_ERR(__wt_calloc_one(session, &fh));
 	WT_ERR(__wt_strdup(session, name, &fh->name));
 
+	/*
+	 * Mark WiredTiger owned files. To keep code simple, we assume that
+	 * the path separator is one char.
+	 */
+	sep = __wt_path_separator();
+	WT_ASSERT(session, strlen(sep) == 1);
+	fp = strrchr(name, sep[0]);
+	if (fp == NULL)
+		fp = name;
+	else
+		++fp;
+	if (WT_PREFIX_MATCH(fp, "WiredTiger"))
+		F_SET(fh, WT_FH_WIREDTIGER_OWNED);
+
```

Comment:
This code surprised me a bit, I was expecting a check for `WT_FS_OPEN_FILE_TYPE_DATA | WT_FS_OPEN_FILE_TYPE_LOG`. How is this different from that test?

Reply:
@ddanderson Please investigate these comments.

Reply:
Thanks, @keithbostic.  Looks like checking `file_type != WT_FS_OPEN_FILE_TYPE_DATA` could be used to determine if a file is owned by WT.  The one counter example I've seen is the `WiredTigerLAS` file, that is marked as `WT_FS_OPEN_FILE_TYPE_DATA`.  So it would participate in file syncing, where it wouldn't with the current code.  That's probably not a bad thing?  I'm running a performance run to see if there's any noticeable affect for my test case.

Reply:
`WiredTigerLAS` has the right prefix, so why doesn't i tmatch? I'm sure I'm missing something terribly obvious...

Regardless, I agree with you that the LAS file should participate in file syncing, there's a lot of traffic to that file in some workloads, and I suspect there will be more in the future.

Reply:
I thought the point of your comment about the file types was to look at the type instead of doing the string comparisons. So doing that would remove the code about matching the prefix.

Reply:
Yes: if we can use the file type flags instead, all of this goes away.

If we can't make that change, then this suggestion might get rid of the path separator assert.

Reply:
@keithbostic, I've pushed a change to use the file type. `WiredTigerLAS` is opened as `WT_FS_OPEN_FILE_TYPE_DATA`, so it will participate in file sync, it didn't before.  I didn't see any change in performance - I guess that's not really surprising, as I don't think we we're doing anything that requires retaining history.

Reply:
I think `WT_FH_WIREDTIGER_OWNED` is confusing. Should we change it to something like `WT_FH_BACKGROUND_SYNC` (and reverse when it's set)?

Reply:
Done.